app/src/main/java/org/merakilearn/ui/onboarding/WelcomeFragment.kt

```diff
@@ -74,6 +80,63 @@ class WelcomeFragment : BaseFragment() {
         })
     }
 
+    private fun initGoogleSignInOption() {
+        val gso =
+            GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestEmail()
+                .requestIdToken(getString(R.string.server_client_id))
+                .build()
+        mGoogleSignInClient = GoogleSignIn.getClient(requireActivity(), gso)
+    }
+
+    private fun signInWithGoogle() {
+        val signInIntent = mGoogleSignInClient!!.signInIntent
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == RC_SIGN_IN) {
+            val task =
+                GoogleSignIn.getSignedInAccountFromIntent(data)
+            handleSignInResult(task)
+        }
+    }
+
+    private fun handleSignInResult(completedTask: Task<GoogleSignInAccount>) {
+        try {
+            val account = completedTask.getResult(ApiException::class.java)
+            logOutSelectedAccount(account)
```

Comment:
why are we signing out from the account in case of sign in?

Reply:
Google save the account selected, and when the user log out from application and try to login again, it doesn't prompt for email selection again and it select the earlier one by default. This behaviour wasn't desired by the team so need to do

Reply:
We should not log out instead try clearing the default account. Logging user out means we won't be able to access token for fetching information. For reference take a loot at this - https://stackoverflow.com/a/39814778/4260786

Reply:
I have implemented this earlier but was causing crash ! I'll try again and let you know